gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -792,6 +800,9 @@ public void flush(String dbName, String tableName) throws IOException {
         String topic = props.get(TOPIC_NAME_KEY);
         if (tableMetadata.appendFiles.isPresent()) {
           tableMetadata.appendFiles.get().commit();
+          if (auditWhitelistBlacklist.acceptTable(dbName, tableName)) {
```

Comment:
There is a chance of optimization here. We should check if a table should be audited when collecting the counts rather than publish time.